stacks/spica/src/storage/components/add-directory-dialog/add-directory-dialog.component.html

```diff
@@ -0,0 +1,27 @@
+<div>
+  <h3>Create A {{ (data.type ? data.type : "folder").replace("_", " ") | titlecase }}</h3>
+  <mat-divider></mat-divider>
+  <mat-form-field>
+    <mat-label>Name</mat-label>
+
+    <input
+      required
+      matInput
+      name="name"
+      [(ngModel)]="name"
+      (ngModelChange)="updateExistsState(nameModel, $event)"
+      #nameModel="ngModel"
+      (keyup.enter)="!nameModel.invalid && dialogRef.close(name)"
+      pattern="^[^\/\ \.]+"
+    />
+    <mat-error *ngIf="nameModel?.errors?.nameExists">
+      This name already exists.
+    </mat-error>
+    <mat-error *ngIf="nameModel?.errors?.pattern">
+      Name must not include slash, space or dot characters.
+    </mat-error>
+  </mat-form-field>
+  <button mat-button [disabled]="nameModel.invalid" (click)="dialogRef.close(name)">
```

Comment:
I didn't see the can-interact directive. Have you tested it without the storage policy?

Reply:
Fixed at [this](https://github.com/spica-engine/spica/pull/919/commits/7f0935d92a4f8e38c290ec6d2407cc93c486c732) commit.